framework/auth/views.py

```diff
@@ -199,87 +199,109 @@ def forgot_password_post(auth, **kwargs):
 
 
 @collect_auth
-def auth_login(auth, **kwargs):
+def auth_register(auth, **kwargs):
     """
-    This view serves as the entry point for OSF login and campaign login.
+    View for the sign-up page. It also serves as the landing page for such services as "prereg" and "claim contributor"
+    where users may not have an OSF account. OSF and institution login options are provided on this page as well.
     HTTP Method: GET
 
-        GET '/login/' without any query parameter:
-            redirect to CAS login page with dashboard as target service
+    :param kwargs: campaign, currently `institution` and `prereg`
+    :param kwargs: next, provided as default CAS service url
+    :param kwargs: logout, for `claim_user_registered` when users don't an account or choose to use another account
+    """
 
-        GET '/login/?logout=true
-            log user out and redirect to CAS login page with redirect_url or next_url as target service
+    data = {}
+    campaign = request.args.get('campaign')
+    service_url = request.args.get('next')
+    log_out = request.args.get('logout')
+    must_login_warning = False
 
-        GET '/login/?campaign=instituion:
-            if user is logged in, redirect to 'dashboard'
-            show institution login
+    # If `?next=[service_url]` is not provided, redirect to `/dashboard`
+    if not service_url:
+        service_url = web_url_for('dashboard', _absolute=True)
 
-        GET '/login/?campaign=prereg:
-            if user is logged in, redirect to prereg home page
-            else show sign up page and notify user to sign in, set next to prereg home page
+    # Only allow redirects which are relative root or full domain, disallows external redirects.
+    if not (service_url[0] == '/' or
+            service_url.startswith(settings.DOMAIN) or
+            service_url.startswith(settings.CAS_SERVER_URL) or
+            service_url.startswith(settings.MFR_SERVER_URL)):
+        raise HTTPError(http.InvalidURL)
 
-        GET '/login/?next=next_url:
-            if user is logged in, redirect to next_url
-            else redirect to CAS login page with next_url as target service
-    """
+    # Campaign
+    if campaign:
+        if campaign in campaigns.CAMPAIGNS:
+            # `GET /register?campaign=prereg`,
+            # Overwrite CAS service url with dedicated campaign url and set `must_login_warn` notification
+            if campaign != 'institution':
```

Comment:
Minor: If it's all the same, I'd go with a positive if statement `if campaign == 'institution':`; it's just a tiny bit less overhead when you're trying to figure out what the code is doing.